tools/cli/src/gear/code.ts

```diff
@@ -0,0 +1,52 @@
+import { CodeChangedData, CodeMetadata, GearApi, generateCodeHash } from '@gear-js/api';
+import { KeyringPair } from '@polkadot/keyring/types';
+import { Option } from '@polkadot/types-codec';
+import fs from 'fs';
+
+import { ICode, SchemeCode } from '../types';
+import { logger } from '../utils';
+
+export async function uploadCode(api: GearApi, account: KeyringPair, pathToProgram: string) {
+  const code = fs.readFileSync(pathToProgram);
+
+  const codeHash = generateCodeHash(code);
+  logger.info(`Code hash: ${codeHash}`, { lvl: 1 });
+
+  const uploadedCode = (await api.query.gearProgram.metadataStorage(codeHash)) as Option<CodeMetadata>;
+
+  if (uploadedCode.isSome) {
+    const blockHash = await api.blocks.getBlockHash(uploadedCode.unwrap().blockNumber);
+    logger.warn(`Code is already uploaded in block ${blockHash}`, { lvl: 1 });
+    return { codeHash, blockHash: blockHash.toHex() };
+  }
+
+  await api.code.upload(code);
+
+  const blockHash: `0x${string}` = await new Promise((resolve, reject) =>
```

Comment:
Would be good to create for this `0x${string}` custom type. Because you have used the type also in other files too.